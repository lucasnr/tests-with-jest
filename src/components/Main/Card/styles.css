.card {
	background-color: #fff;
	border-radius: 0.5rem;
	display: flex;
	flex-direction: column;
	padding: 2rem;
}

.card .card__title {
	color: #d44;
	font-size: 1.25rem;
	font-weight: 600;
	margin-bottom: 1rem;
}

.card .card__text {
	font-size: 0.875rem;
	font-weight: 400;
	line-height: 1.5;
}
