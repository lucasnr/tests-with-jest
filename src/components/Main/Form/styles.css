#main .form {
	display: flex;
	flex-direction: column;
	padding: 2rem;
	padding-bottom: 0px;
	width: 100%;
}

#main .form > *:not(:last-child) {
	margin-bottom: 1.5rem;
}

#main .form input,
#main .form textarea {
	background-color: #fff;
	border: none;
	font-size: 0.875rem;
	padding: 0.625rem 1.125rem;
	width: 100%;
}

#main .form input {
	border-radius: 9999px;
}

#main .form textarea {
	border-radius: 0.5rem;
}

#main .form button {
	background-color: #d44;
	border: 0.0625rem solid #d44;
	border-radius: 9999px;
	color: #fff;
	font-size: 0.875rem;
	padding: 0.625rem 1.125rem;
	text-align: center;
	transition: background-color 0.2s ease, color 0.2s ease;
}

#main .form button:hover,
#main .form button:focus {
	background: none;
	color: #d44;
}

@media (min-width: 768px) {
	#main .form {
		align-items: flex-start;
	}
}
