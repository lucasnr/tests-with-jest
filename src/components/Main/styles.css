#main {
	background-color: #eee;
	flex-grow: 1;
}

#main .cards {
	display: grid;
	grid-gap: 1.5rem;
	padding: 2rem;
	width: 100%;
}

@media (min-width: 576px) {
	#main .cards {
		grid-template-columns: repeat(2, 1fr);
	}
}

@media (min-width: 768px) {
	#main .cards {
		grid-template-columns: repeat(3, 1fr);
	}
}

@media (min-width: 1200px) {
	#main .cards {
		grid-template-columns: repeat(4, 1fr);
	}
}
