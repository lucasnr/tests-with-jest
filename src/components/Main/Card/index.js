import React from 'react';

import './styles.css';

function Card({ title, text }) {
	return (
		<li className="card">
			<h2 className="card__title">{title}</h2>
			<p className="card__text">{text}</p>
		</li>
	);
}

export default Card;
