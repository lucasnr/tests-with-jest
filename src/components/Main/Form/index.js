import React, { useCallback, useState } from 'react';

import './styles.css';

function Form({ onSubmit }) {
	const [title, setTitle] = useState();
	const [text, setText] = useState();
	const handleSubmit = useCallback(
		(event) => {
			event.preventDefault();

			const data = {
				title,
				text,
			};

			onSubmit(data);
		},
		[onSubmit, text, title]
	);

	return (
		<form className="form" onSubmit={handleSubmit}>
			<input
				name="title"
				type="text"
				placeholder="Enter the title"
				required
				value={title || ''}
				onChange={(event) => setTitle(event.target.value)}
			/>
			<textarea
				name="text"
				cols="30"
				rows="10"
				placeholder="Enter the text"
				required
				value={text || ''}
				onChange={(event) => setText(event.target.value)}
			/>

			<button type="submit">Add new card</button>
		</form>
	);
}

export default Form;
