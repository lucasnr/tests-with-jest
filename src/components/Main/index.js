import React, { useState, useCallback } from 'react';

import Form from './Form';
import Card from './Card';
import './styles.css';

function Main() {
	const [cards, setCards] = useState([]);

	const handleSubmit = useCallback((data) => {
		const { title, text } = data;

		setCards((prev) => [
			...prev,
			{
				title,
				text,
			},
		]);
	}, []);

	return (
		<main id="main">
			<Form onSubmit={handleSubmit} />

			<ul className="cards">
				{cards.length > 0 &&
					cards.map((card, index) => <Card key={index} {...card} />)}
			</ul>
		</main>
	);
}

export default Main;
