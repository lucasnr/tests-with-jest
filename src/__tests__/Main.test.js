import React from 'react';
import { shallow, mount } from 'enzyme';
import Main from '../components/Main';

import Form from '../components/Main/Form';
import Card from '../components/Main/Card';

describe('Main', () => {
	it('renders without crashing and matches snapshot', () => {
		const wrapper = shallow(<Main />);
		expect(wrapper).toMatchSnapshot();
	});

	it('renders card on form submit', () => {
		const wrapper = mount(<Main />);

		wrapper.find(Form).simulate('submit');
		expect(wrapper.find(Card).length).toBe(1);

		wrapper.find(Form).simulate('submit');
		expect(wrapper.find(Card).length).toBe(2);
	});

	it('renders card with form data', () => {
		const wrapper = mount(<Main />);
		const form = wrapper.find('.form');

		const title = 'Title defined by enzyme';
		const input = form.find('input[name="title"]');
		input.simulate('change', {
			target: { value: title },
		});

		const text = 'Text defined by enzyme';
		const textarea = form.find('textarea[name="text"]');
		textarea.simulate('change', {
			target: { value: text },
		});

		form.simulate('submit');

		const card = wrapper.find(Card);
		expect(card.find('.card__title').text()).toBe(title);
		expect(card.find('.card__text').text()).toBe(text);
	});
});
