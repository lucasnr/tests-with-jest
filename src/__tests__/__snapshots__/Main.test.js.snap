// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders without crashing and matches snapshot 1`] = `
<main
  id="main"
>
  <Form
    onSubmit={[Function]}
  />
  <ul
    className="cards"
  />
</main>
`;
